Minorfixes to "new" gtm_stringread() functions to eliminate leading and trailing spaces

robertlipe opened the pull request.

robertlipe committed:
Trim more uses of C date and potentially unsafe memory practices.

robertlipe committed:
GTM: new fread_string() now whackes leading and trailing space

tsteven4 commented on dmtlog.cc:
```diff
@@ -425,8 +418,8 @@ read_CTrackFile(const int version)
-    char* s = read_str(fin);
-    xfree(s);
+    auto s = read_str(fin);
+    Q_UNUSED(s);
```
or without Qtisms"(void) read_str(fin);"

tsteven4 commented on gtm.cc:
```diff
@@ -79,20 +79,8 @@ static QString
-  xfree(val);
-  return v;
+  QByteArray bytes = gbfreadbuf(len, fd);
+  return QString(bytes).trimmed();
```
or "return gbfreadbuf(len, fd).trimmed();"
QByteArray also has trimmed, and it's not necessary to use the QString constructor explicitly.

tsteven4 commented on gtm.cc:
```diff
@@ -104,17 +92,8 @@ fread_string_discard(gbfile* fd)
-
-  return v;
+  QByteArray bytes = gbfreadbuf(len, fd);
+  return QString(bytes);
```
or "return gbfreadbuf(len, fd);"

tsteven4 commented on dmtlog.cc:
```diff
@@ -329,21 +330,16 @@ tlog3b_xgcb_wpten(xg_string, const QXmlStreamAttributes*)
-
-  return res;
+  auto res = gbfreadbuf(i, f);
+  return QString(res);
```
or "return gbfreadbuf(i, f);", no need to call QString constructor.

tsteven4 commented on vitosmt.cc:
```diff
@@ -128,20 +125,16 @@ vitosmt_read()
     double usec = fmod(1000000*seconds+0.5,1000000);
-    wpt_tmp->SetCreationTime(mkgmtime(&tmStruct), lround(usec/1000.0));
+    QTime wtime(timestamp[3], timestamp[4],
+                (int) floor(seconds), lround(usec/1000.0));
```
This is problematic if usec/1000.0 rounds up to 1000, in that case wtime will be invalid.

tsteven4 commented on vitosmt.cc:
```diff
@@ -128,20 +125,16 @@ vitosmt_read()
-    tmStruct.tm_isdst =-1;
+    QDate wdate(timestamp[0] + 2000, timestamp[1], timestamp[2]);
 
     double usec = fmod(1000000*seconds+0.5,1000000);
```
This is problematic.  It can result in incorrect rounding, e.g.
seconds: 59.9999999 usec: 0.3999999985098839

tsteven4 commented on vitosmt.cc:
```diff
@@ -128,20 +125,16 @@ vitosmt_read()
-    wpt_tmp->SetCreationTime(mkgmtime(&tmStruct), lround(usec/1000.0));
+    QTime wtime(timestamp[3], timestamp[4],
+                (int) floor(seconds), lround(usec/1000.0));
+    wpt_tmp->SetCreationTime(QDateTime(wdate, wtime, Qt::UTC));
```
I suggest something like:
wpt_tmp->SetCreationTime(QDateTime(QDate((timestamp[0] + 2000, timestamp[1], timestamp[2]), QTime(timestamp[3], timestamp[4]), Qt::UTC).addMSecs(lround(seconds*1000.0));

robertlipe committed:
Integrate Stephen's suggestions. (I wrote that half asleep and kept
making it worse..sorry)
More Qt6-proofing: Add fixes for deprecation warnings on endl and hex.

robertlipe committed:
Add warning fixes for garmin_xt and exif

robertlipe committed:
Forget about fixing warning on nmea's use of hex. Revert.

robertlipe committed:
And revert use ot Qt::endl
